### PR TITLE
[CDAP-13100] Add BigQuery Connection

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-13100

Backend Service: https://github.com/hydrator/wrangler/pull/205

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/BigQueryBrowser.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+@import '../../../../styles/variables.scss';
+
+$header-border: #cccccc;
+$header-bg: #eeeeee;
+$content-font-color: #333333;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/BigQueryBrowser.scss]: We can set these variables to the various `$grey-` values we have in `variables.scss`

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/BigQueryBrowser.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+.bigquery-browser {
+  height: calc(100vh - 50px - 54px);
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/BigQueryBrowser.scss]: nit: Maybe add a comment to describe what these px values are?

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/DisplaySwitch/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+DisplaySwitchView.propTypes = {
+  datasetId: PropTypes.string,
+  onWorkspaceCreate: PropTypes.func
+};
+
+const mapStateToProps = (state) => {
+  return {
+    datasetId: state.bigquery.datasetId
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/DisplaySwitch/index.js:42]: You're using `onWorkspaceCreate` inside `DisplaySwitchView`, but you're not passing `onWorkspaceCreate` as prop?

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/TableList/index.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    let params = {
+      namespace,
+      connectionId: this.props.connectionId,
+      datasetId: this.props.datasetId,
+      tableId
+    };
+
+    MyDataPrepApi.readBigQueryTable(params)
+      .subscribe((res) => {
+        let workspaceId = res.values[0].id;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/TableList/index.js]: nit: use objectQuery?

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js [hunk 5]
```diff
@@ -335,6 +335,47 @@ export default class AddToHydratorModal extends Component {
     };
   }
 
+  constructBigQuerySource(artifactsList, bigqueryInfo) {
+    if (!bigqueryInfo) { return null; }
+
+    let batchArtifact = find(artifactsList, {name: 'google-cloud'});
+    if (!batchArtifact) {
+      this.setState({
+        error: T.translate(`${PREFIX}.BigQueryPipeline.pipelineError`),
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js]: seems like you haven't actually added `BigQueryPipeline` under `DataPrep.TopPanel` in text-en.yaml?

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js [hunk 6]
```diff
@@ -296,6 +296,43 @@ function constructGCSSource(artifactsList, gcsInfo) {
   };
 }
 
+function constructBigQuerySource(artifactsList, bigqueryInfo) {
+  if (!bigqueryInfo) { return null; }
+
+  let batchArtifact = find(artifactsList, {name: 'google-cloud'});
+  if (!batchArtifact) {
+    T.translate(`features.DataPrep.TopPanel.BigQueryPipeline.ingestDataerror`);
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js]: Same here about i18n. Also, don't you need a `return`?

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js [hunk 7]
```diff
@@ -335,6 +335,47 @@ export default class AddToHydratorModal extends Component {
     };
   }
 
+  constructBigQuerySource(artifactsList, bigqueryInfo) {
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js]: Is the only difference between this function and the one in `PipelineConfigHelper` the error handling for `if (!batchArtifact)`? Can/Should we try to combine them?

File: cdap-ui/app/cdap/components/DataPrepConnections/BigQueryConnection/BigQueryConnection.scss [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+$asterisk-color: #d50000;
+$success-color: #00c853;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrepConnections/BigQueryConnection/BigQueryConnection.scss:18]: Seems like these two colors are used in all the `.scss` files for DataPrep connections. Maybe we can move them to `variables.scss`?

File: cdap-ui/app/cdap/components/DataPrepConnections/index.js [hunk 9]
```diff
@@ -399,6 +411,40 @@ export default class DataPrepConnections extends Component {
     );
   }
 
+  renderBigQueryDetail() {
+    let namespace = NamespaceStore.getState().selectedNamespace;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrepConnections/index.js]: nit: just import `getCurrentNamespace()` directly

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 10]
```diff
@@ -337,6 +337,14 @@ features:
       suggestion1: the page
       suggestionTitle: "You can try to:"
     DataPrepBrowser:
+      BigQueryBrowser:
+        title: Select Table
+        datasets: Datasets
+        name: Name
+        datasetCount:
```
Comment (tonybach) [inline cdap-ui/app/cdap/text/text-en.yaml]: nit: sort these?

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/DisplaySwitch/index.js (hunk 3 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/BigQueryBrowser/DisplaySwitch/index.js:42]: It is being passed as prop, from `BigQueryBrowser` class render method

File: cdap-ui/app/cdap/components/DataPrepConnections/BigQueryConnection/BigQueryConnection.scss (hunk 8 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/DataPrepConnections/BigQueryConnection/BigQueryConnection.scss:18]: I will refactor this later. The entire DataprepConnections modals styling can be consolidated.

**Comment by tonybach** [comment]
Code LGTM 👍

**Review by bdmogal — APPROVED** [review]

**Pull request merged**